server/src/main/java/org/apache/druid/server/coordinator/helper/NewestSegmentFirstIterator.java

```diff
@@ -354,25 +358,31 @@ private static Interval findInitialSearchInterval(
 
     final Interval skipInterval = new Interval(skipOffset, last.getInterval().getEnd());
 
-    final List<TimelineObjectHolder<String, DataSegment>> holders = timeline.lookup(
-        new Interval(first.getInterval().getStart(), last.getInterval().getEnd().minus(skipOffset))
-    );
+    final DateTime lookupStart = first.getInterval().getStart();
+    final DateTime lookupEnd = last.getInterval().getEnd().minus(skipOffset);
+    if (lookupStart.isBefore(lookupEnd)) {
+      final List<TimelineObjectHolder<String, DataSegment>> holders = timeline.lookup(
+          new Interval(first.getInterval().getStart(), last.getInterval().getEnd().minus(skipOffset))
```

Comment:
Change to `new Interval(lookupStart, lookupEnd)`.

Reply:
Thanks. Fixed.